src/rpc/methods/msig.rs

```diff
@@ -0,0 +1,181 @@
+            )),
+            std::cmp::Ordering::Equal => Ok(BigInt::from(0)),
+            std::cmp::Ordering::Less => {
+                let msig_actor = ctx
+                    .state_manager
+                    .get_required_actor(&addr, *end_ts.parent_state())?;
+                let ms = multisig::State::load(ctx.store(), msig_actor.code, msig_actor.state)?;
+                let start_lb: TokenAmount = ms.locked_balance(start_ts.epoch())?.into();
+                let end_lb: TokenAmount = ms.locked_balance(end_ts.epoch())?.into();
+                Ok(start_lb.atto() - end_lb.atto())
+            }
+        }
+    }
+}
+
+pub enum MsigGetVestingSchedule {}
+impl RpcMethod<2> for MsigGetVestingSchedule {
+    const NAME: &'static str = "Filecoin.MsigGetVestingSchedule";
+    const PARAM_NAMES: [&'static str; 2] = ["address", "tsk"];
+    const API_VERSION: ApiVersion = ApiVersion::V0;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = (Address, ApiTipsetKey);
+    type Ok = MsigVesting;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (addr, ApiTipsetKey(tsk)): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let ts = ctx.chain_store.load_required_tipset_or_heaviest(&tsk)?;
+
+        let msig_actor = ctx
+            .state_manager
+            .get_required_actor(&addr, *ts.parent_state())?;
+        let ms = multisig::State::load(ctx.store(), msig_actor.code, msig_actor.state)?;
+
+        let msig_vesting = match &ms {
+            multisig::State::V8(st) => MsigVesting {
+                initial_balance: st.initial_balance.atto().clone(),
+                start_epoch: st.start_epoch,
+                unlock_duration: st.unlock_duration,
+            },
+            multisig::State::V9(st) => MsigVesting {
+                initial_balance: st.initial_balance.atto().clone(),
+                start_epoch: st.start_epoch,
+                unlock_duration: st.unlock_duration,
+            },
+            multisig::State::V10(st) => MsigVesting {
+                initial_balance: st.initial_balance.atto().clone(),
+                start_epoch: st.start_epoch,
+                unlock_duration: st.unlock_duration,
+            },
+            multisig::State::V11(st) => MsigVesting {
+                initial_balance: st.initial_balance.atto().clone(),
+                start_epoch: st.start_epoch,
+                unlock_duration: st.unlock_duration,
+            },
+            multisig::State::V12(st) => MsigVesting {
+                initial_balance: st.initial_balance.atto().clone(),
+                start_epoch: st.start_epoch,
+                unlock_duration: st.unlock_duration,
+            },
+            multisig::State::V13(st) => MsigVesting {
+                initial_balance: st.initial_balance.atto().clone(),
+                start_epoch: st.start_epoch,
+                unlock_duration: st.unlock_duration,
+            },
+        };
```

Comment:
Is this a good place for this? @hanabi1224 implemented it `src/shim/actors/market/balance_table.rs`, which allows re-usability.

Reply:
Ah, we have a custom struct here. Then perhaps it belongs to `forest`. 

The stuff I was talking about before: 

https://github.com/ChainSafe/fil-actor-states/blob/main/fil_actor_interface/src/builtin/market/mod.rs#L171

Reply:
moved inside `src/shim/actors/multisig`